### PR TITLE
Feature/cax 1 prs@152 broker api

### PR DESCRIPTION
**Pull request by algattik**
- Broker API implementation. 
- Unit test coverage.

### CODE REVIEW

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/controllers/BrokerProxyController.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    @Operation(operationId = "uploadPartRelationshipUpdateList",
+         summary = "Upload a PartRelationshipUpdateList. " + PartRelationshipUpdateRequest.DESCRIPTION)
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "204" /* no content */,
+            description = "PartRelationshipUpdateList uploaded successfully"),
+        @ApiResponse(responseCode = "400", description = "Bad request",
+                content = {@Content(mediaType = APPLICATION_JSON_VALUE,
+                        schema = @Schema(implementation = ErrorResponse.class))}),
+    })
+    @PostMapping("/PartRelationshipUpdateList")
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/controllers/BrokerProxyController.java:62]: Uppercase in mappings is quite uncommon...

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/annotations/ValueOfEnum.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Custom annotation to validate input for enum.
+ */
+@Target({FIELD})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = ValueOfEnumValidator.class)
+@SuppressWarnings("checkstyle:MissingJavadocMethod")
+public @interface ValueOfEnum {
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/annotations/ValueOfEnum.java:31]: Why there is no common package for the Query API and the Broker Proxy? No need to change anything in this PR, just wondering if there was any reasoning behind it.

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/requests/PartAttributeUpdateRequest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import javax.validation.constraints.NotNull;
+import java.time.Instant;
+
+/*** Request for updates to {@link PartInfo}s. */
+@Schema(description = PartAttributeUpdateRequest.DESCRIPTION)
+@Value
+@Builder(toBuilder = true, setterPrefix = "with")
+@JsonDeserialize(builder = PartAttributeUpdateRequest.PartAttributeUpdateRequestBuilder.class)
+@SuppressWarnings("PMD.CommentRequired")
+public class PartAttributeUpdateRequest {
```
[the remark this answers is not in the document]
Comment (marcgs) [reply coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/requests/PartAttributeUpdateRequest.java:31]: Let's discuss what the easiest way forward here is. I see 3 options:
-  We implement the Broker API with the same interface we communicated to the test generator so we can reuse it without any manipulation on our side
- We inform the test data generator team of this change and try to get them implement the change on their side
- We write a script that transforms what the generator outputs into this new format

[the remark this answers is not in the document]
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/requests/PartAttributeUpdateRequest.java:31]: As per DB and latest design we created it like this to keep it generic for any attribute. So may be we can inform the test data generator team if they accommodate our change easily.

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/annotations/ValueOfEnum.java (hunk 2 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/annotations/ValueOfEnum.java:31]: No strong reason, just it was like this but I will take a look on your point and see what we can do here.

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/requests/PartAttributeUpdateRequest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    @NotNull
+    @Valid
+    @Schema(implementation = PartId.class)
+    private PartId part;
+
+    @NotNull
+    @ValueOfEnum(enumClass = PartAttributeName.class)
+    @Schema(description = "Attribute name")
+    private String name;
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/requests/PartAttributeUpdateRequest.java:42]: Is it inlined with PartAttributeEntity? In PartAttributeEntity we have fields representing attribute value and part name. And here _name_ and _value_ are related to the attribute. What's the difference between _name_ and _value_ ?

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/controllers/BrokerProxyController.java (hunk 1 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/controllers/BrokerProxyController.java]: Updated to start with lowercase.

**Review by marcgs — APPROVED** [review]

**Pull request merged**